### DISCUSSION TITLE
Setting response schema for c.json()

### DISCUSSION THREAD

**Discussion by kofta999** [honojs/hono#2655] [Q&A] [answered]
Hello, I'm new to Hono and I was wondering if there's a way to set a response schema to c.json() or to set a global schema for all json responses

I found out I could use
```ts
interface CustomResponse {
  success: boolean;
  message: string;
}

return c.json<CustomResponse, 400>({ success: false, message: "error" }, 400);
```

I don't mind typing the interface every time but retyping the status code doesn't seem a good idea, and if I removed status code from the generic TypeScript throws an error, and if I removed it from the parameters it just returns the default 200.

**Comment by yusukebe** [comment, marked as answer]
Hi @kofta999 

Unfortunately, we don't have a public API to specify response Types explicitly, so there is nothing to do about your question.

**Reply by kofta999** [reply]
Understood, thank you for answering!

**Reply by yusukebe** [reply]
Regarding validating a response, we have some idea, though it's not implemented: https://github.com/honojs/hono/issues/2439

**Reply by kofta999** [reply]
Thank you, got the main idea of it, but what I was thinking about was more on the type level than the runtime, just to easily type my response fields for better DX.

Will probably then type it using the generics like I did before. Will also share if I found a better option. Thanks again!
